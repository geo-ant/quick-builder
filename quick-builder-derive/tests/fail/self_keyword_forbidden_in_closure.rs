use quick_builder_derive::QuickBuilder;

#[derive(QuickBuilder)]
struct Foo {
    #[invariant(|val|{
        {
           if *val>0. {
               // we hide the usage away in the AST to make sure the
               // parsing works correctly
               let f = self;
           }
        }
        *val>0.})]
    x: f32,
}

fn main() {}
