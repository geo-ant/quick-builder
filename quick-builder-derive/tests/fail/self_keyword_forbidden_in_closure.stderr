error: self keyword not allowed in validation closure
  --> tests/fail/self_keyword_forbidden_in_closure.rs:10:24
   |
10 |                let f = self;
   |                        ^^^^
